### PR TITLE
[go-metro] Adding supervisor entry and sample yaml config.

### PR DESCRIPTION
**Pull request by truthbk**
Alternatively, we could add a python launcher that will only launch the binary iff `conf.d/dd-tcp-rtt.yaml` exists.

### CODE REVIEW

File: conf.d/go-metro.yaml.example [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# NOTE: for go metro to run unprivileged, you will have to set CAP_NET_RAW capabilities on the binary:
+#        (debian based) $ sudo apt-get update libcap
+#        (redhat based) $ sudo yum install compat-libcap1
+#        (redhat based - alternatively) $ sudo yum install compat-libcap1
```
Comment (olivielpeau) [inline conf.d/go-metro.yaml.example]: The alternative command is the same as the regular one ;)

Comment (olivielpeau) [inline conf.d/go-metro.yaml.example]: Unrelated: let's add a comment to mention that `go-metro` logs to a separate file

File: utils/flare.py [hunk 2]
```diff
@@ -215,6 +216,7 @@ def _add_logs_tar(self):
         self._add_log_file_tar(self._forwarder_log)
         self._add_log_file_tar(self._dogstatsd_log)
         self._add_log_file_tar(self._jmxfetch_log)
+        self._add_log_file_tar(self._gometro_log)
```
Comment (olivielpeau) [inline utils/flare.py]: by default we're going to log a `warning` if the log file is not present. That could confuse users if they're running anything else than a packaged Linux agent...

File: conf.d/go-metro.yaml.example (hunk 1 above)
Comment (truthbk) [inline conf.d/go-metro.yaml.example]: Are you sure it's the exact same `rpm`? I was expecting yum to know they were aliases, but it considered them distinct.

File: conf.d/go-metro.yaml.example [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# NOTE: for go metro to run unprivileged, you will have to set CAP_NET_RAW capabilities on the binary:
+#        (debian based) $ sudo apt-get update libcap
+#        (redhat based) $ sudo yum install libcap
+#        (redhat based - alternatively) $ sudo yum install compat-libcap1
+#        $ sudo setcap cap_net_raw+ep /opt/datadog-agent/bin/go-metro
+#
+#        Also, please note that go-metro logs to its own file - found in /var/log/datadog/go-metro.log.
```
Comment (olivielpeau) [inline conf.d/go-metro.yaml.example]: Last thing we could mention here: `go-metro` does _not_ appear as a check on the agent's `info` page (IMO this is counter-intuitive for users used to regular checks/JMXFetch checks)

File: utils/flare.py [hunk 4]
```diff
@@ -215,13 +216,14 @@ def _add_logs_tar(self):
...
         self._add_log_file_tar(self._dogstatsd_log)
         self._add_log_file_tar(self._jmxfetch_log)
+        self._add_log_file_tar(self._gometro_log, warn=False)
         self._add_log_file_tar(
             "{0}/*supervisord.log".format(os.path.dirname(self._collector_log))
         )
 
-    def _add_log_file_tar(self, file_path):
+    def _add_log_file_tar(self, file_path, warn=True):
         for f in glob.glob('{0}*'.format(file_path)):
```
Comment (olivielpeau) [inline utils/flare.py]: actually the `glob.glob` step here prevents us from adding a log file that doesn't exist, so you can ignore my comment on the warnings and revert these changes, thanks!

File: packaging/supervisor.conf [hunk 5]
```diff
@@ -57,5 +57,14 @@ priority=999
 startsecs=3
 user=dd-agent
 
+[program:go-metro]
+command=/opt/datadog-agent/bin/go-metro -cfg="/etc/dd-agent/conf.d/go-metro.yaml"
+stdout_logfile=NONE
+stderr_logfile=NONE
+redirect_stderr=true
+priority=999
+startsecs=3
```
Comment (olivielpeau) [inline packaging/supervisor.conf]: Let's lower this to 2 secs, and lower the number of seconds in go-metro too

File: utils/flare.py (hunk 2 above)
Comment (truthbk) [inline utils/flare.py]: we're not really going to log a warning ;)

File: conf.d/go-metro.yaml.example (hunk 3 above)
Comment (truthbk) [inline conf.d/go-metro.yaml.example]: added below.

**Comment by olivielpeau** [comment]
:+1:
